core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -1114,24 +1114,22 @@ class LogTest extends JUnitSuite {
   }
 
   @Test
-  def shouldDeleteTimeBasedSegmentsReadyToBeDeleted(): Unit = {
-    val set = TestUtils.singleMessageSet("test".getBytes)
+  def shouldDeleteTimeBasedSegmentsReadyToBeDeleted() {
+    val set = TestUtils.singleMessageSet("test".getBytes, timestamp = 10)
     val (myTime, log) = createLog(set.sizeInBytes, retentionMs = 10000)
 
     // append some messages to create some segments
     for (i <- 0 until 15)
       log.append(set)
 
-    log.logSegments.head.lastModified = myTime.milliseconds - 20000
-
     log.deleteOldSegments()
-    assertEquals("There should be 2 segments remaining", 2, log.numberOfSegments)
+    assertEquals("There should be 1 segment remaining", 1, log.numberOfSegments)
```

Comment:
Yes, because i changed it to use a message timestamp. The timestamp on the messages are all set to 10 and the retentionMs is set to 10000, so there should only be 1 segment remaining